test/verify/check-machines

```diff
@@ -30,42 +30,127 @@ def readFile(name):
             content = f.read().replace('\n', '')
     return content
 
+SPICE_XML="""
+    <video>
+      <model type='vga' heads='1' primary='yes'/>
+      <alias name='video0'/>
+    </video>
+    <graphics type='spice' port='5900' autoport='yes' listen='127.0.0.1'>
+      <listen type='address' address='127.0.0.1'/>
+      <image compression='off'/>
+    </graphics>
+"""
+
+VNC_XML="""
+    <video>
+      <model type='vga' heads='1' primary='yes'/>
+      <alias name='video0'/>
+    </video>
+    <graphics type='vnc' port='5900' autoport='yes' listen='127.0.0.1'>
+      <listen type='address' address='127.0.0.1'/>
+    </graphics>
+"""
+
+CONSOLE_XML="""
+    <console type='file'>
+      <target type='serial' port='0'/>
+      <source path='{log}'/>
+    </console>
+"""
+
+DOMAIN_XML="""
+<domain type='kvm'>
+  <name>{name}</name>
+  <vcpu>1</vcpu>
+  <os>
+    <type arch='x86_64'>hvm</type>
+    <boot dev='hd'/>
+    <boot dev='network'/>
+  </os>
+  <memory unit='MiB'>256</memory>
+  <currentMemory unit='MiB'>256</currentMemory>
+  <features>
+    <acpi/>
+  </features>
+  <devices>
+    <disk type='block' device='disk'>
+      <driver name='qemu' type='raw'/>
+      <source dev='/dev/{disk}'/>
+      <target dev='hda' bus='ide'/>
+      <serial>ROOT</serial>
+    </disk>
+    <disk type='file' snapshot='external'>
+      <driver name='qemu' type='qcow2'/>
+      <source file='{image}'/>
+      <target dev='hdb' bus='ide'/>
+      <serial>SECOND</serial>
+    </disk>
+    <controller type='scsi' model='virtio-scsi' index='0' id='hot'/>
+    <interface type='network'>
+      <source network='default' bridge='virbr0'/>
+      <target dev='vnet0'/>
+    </interface>
+    {console}
+    {graphics}
+  </devices>
+</domain>
+"""
+
 # If this test fails to run, the host machine needs:
 # echo "options kvm-intel nested=1" > /etc/modprobe.d/kvm-intel.conf
 # rmmod kvm-intel && modprobe kvm-intel || true
 
 @skipImage("Atomic cannot run virtual machines", "fedora-atomic", "rhel-atomic", "continuous-atomic")
-@skipImage("Debian Stable creates paused machines", "debian-stable")
 class TestMachines(MachineCase):
-    def startVm(self, name, console='spice'):
+    def startVm(self, name, graphics='spice'):
         m = self.machine
 
+        image_file = m.pull("cirros")
```

Comment:
This looks like it would actually try and pull the cirros image from within the test VM. This would contradict the efforts of making the tests work offline (with using the qemu socket network), and also makes the test take awfully long and unstable. It might be more elegant to just pass the test VM its own image as a block device, which it then can boot nested? Then we also don't need to worry about its architecture any more.